scripts/cache-inference/cache-inference.ts

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Walrus Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import mdbookOperatorsJson from '../../docs/book/assets/operators.json';
+import { AggregatorData, AggregatorDataVerbose, HeaderValue, Operators } from './types';
+
+const KnownCacheKeys = [
+    "cdn-cache",
+    "cdn-cachedat",
+    "cache-status",
+    "cache-control",
+    "cf-cache-status",
+    "x-cache-status"
+];
+
+type HeaderMatch = {
+    key: string;
+    value: string | null
+};
+type HasCacheOutput = {
+    hasCache: boolean;
+    matches: HeaderMatch[];
+};
+
+function headerKeyContainsCache(headers: Headers): HasCacheOutput {
+    const matches = [...headers.entries()].filter(([key, _]) => {
+        return key.toLowerCase().includes("cache")
+    }).map(([key, value]) => { return { key, value } });
+
+    return {
+        hasCache: matches.length > 0,
+        matches
+    };
+}
+
+function headersHaveCacheHit(matches: HeaderMatch[]): boolean {
+    return matches.some(({ key: _, value }) => {
+        return value?.toLowerCase().includes("hit");
+    });
+}
+
+async function updateAggregatorCacheInfo(
+    aggregators: Record<string, AggregatorData>,
+    blobId: string,
+    threshold: number,
+) {
+
+    // Used for debugging purposes
+    const aggregatorsVerbose: Record<string, AggregatorDataVerbose> = {};
+    for (const [url, value] of Object.entries(aggregators)) {
+        const blobUrl = new URL(`v1/blobs/${blobId}`, url);
+        let fetch1: number;
+        let fetch2: number;
+        let headers1: Headers;
+        let headers2: Headers;
+        try {
+            let start = Date.now();
+            const resp1 = await fetch(blobUrl);
+            // Measure the full response time (note though that we should use small blobs anyway
+            // here).
+            await resp1.arrayBuffer();
+            fetch1 = Date.now() - start;
+            headers1 = resp1.headers;
+            start = Date.now();
+            const resp2 = await fetch(blobUrl);
+            await resp2.arrayBuffer();
+            fetch2 = Date.now() - start
+            headers2 = resp2.headers;
+        } catch (e) {
+            console.error(`Error during measuring ${blobUrl}:`);
+            console.error(e);
+            continue;
+        }
+        const speedupMs = fetch1 - fetch2;
+
+        let output1 = headerKeyContainsCache(headers1);
+        let output2 = headerKeyContainsCache(headers2);
```

Comment:
Nit: These could be `const` as well:
```suggestion
        const output1 = headerKeyContainsCache(headers1);
        const output2 = headerKeyContainsCache(headers2);
```
Also, the variable names are not very informative.